File: July-LeetCoding-Challenge/03-Prison-Cells-After-N-Days/Prison-Cells-After-N-Days.cs
public class Solution {
    public int[] PrisonAfterNDays(int[] cells, int N) {
        var length = cells.Length;
        var startCells = new int[length];
        var runningCells = new int[length];       
        for(int i=0; N-->0; cells=runningCells.ToArray(), ++i){
            for(int x=1; x<length-1; x++){
                runningCells[x] = (cells[x-1] == cells[x+1]) ? 1 : 0;
            }
            if(i==0) {
                startCells = runningCells.ToArray();
            }
            else if(startCells.SequenceEqual(runningCells)){
                N %= i;
            }
        }
        return runningCells;
    }
}

File: July-LeetCoding-Challenge/03-Prison-Cells-After-N-Days/Prison-Cells-After-N-Days.cpp
class Solution {
public:
    vector<int> prisonAfterNDays(vector<int>& cells, int N) {
        // vector<vector<int>> memo;
        //memo.push_back(cells);
        N=(N%14==0)?14:N%14;
        for(int i=1; i<=N; ++i){
            vector<int> cpy = cells;
            for(int j=0; j<8; ++j){
                if(j-1>=0&&j+1<8&&cells[j-1]==cells[j+1]) cpy[j]=1;
                else cpy[j]=0;
            }
            /*
            for(int j=0; j<memo.size(); ++j)
                if(memo[j] == cpy)
                    cout<<"day "<< i<<" equals to day "<<j<<endl;
            memo.push_back(cpy);
            */
            swap(cells, cpy);
        }
        return cells;
    }
};
